pkg/config/network.go

```diff
@@ -1,71 +1,165 @@
 package config
 
-type AppConfig struct {
-	Network    string
-	NodeURL    string
-	DNSAddress string
-}
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"sync"
+
+	"gopkg.in/yaml.v2"
+)
 
 const (
-	testnetNodeURL = "https://test.massa.net/api/v2"
-	// testnet20.2.
-	testnetDNSAddress = "AS12YMz7NjyP3aeEWcSsiC58Hba8UxHapfGv7i4PmNMS2eKfmaqqC"
+	MassaStationURL = "station.massa"
+)
 
-	labnetNodeURL    = "https://labnet.massa.net/api/v2"
-	labnetDNSAddress = "AS1PV17jWkbUs7mfXsn8Xfs9AK6tHiJoxuGu7RySFMV8GYdMeUSh"
+// NetworkManager represents a manager for network configurations.
+type NetworkManager struct {
+	appConfig     AppConfig                // Current network configuration
+	knownNetworks map[string]NetworkConfig // Known network configurations
+	networkMutex  sync.RWMutex             // Mutex for concurrent access to network configuration
+}
 
-	buildnetNodeURL    = "https://buildernet.massa.net/api/v2"
-	buildnetDNSAddress = "AS12aGWkBorEM2EpKeNyigSkoCqwdxm872g5KkzUiox3v4VosFW3F"
+// Verify at compilation time that NetworkManager implements NetworkManagerInterface.
+//nolint: exhaustruct
+var _ NetworkManagerInterface = &NetworkManager{}
+
+// NewNetworkManager creates a new instance of NetworkManager.
+// It loads the initial network configurations from the specified file and sets the default network configuration.
+// configFile: The path to the YAML configuration file.
+// Returns the initialized NetworkManager and any error encountered during initialization.
+func NewNetworkManager(configFile string) (*NetworkManager, error) {
+	//nolint: exhaustruct
+	networkManager := &NetworkManager{
+		appConfig:     AppConfig{},
+		knownNetworks: make(map[string]NetworkConfig),
+	}
 
-	MassaStationURL = "station.massa"
-)
+	// Load network configuration from file
+	initNetworksData, err := LoadConfig(configFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load configuration: %w", err)
+	}
+
+	networkManager.SetNetworks(initNetworksData)
+
+	var defaultNetwork string
+
+	for networkName, networkConfig := range initNetworksData {
+		if networkConfig.Default {
+			defaultNetwork = networkName
+
+			break
+		}
+	}
+
+	if defaultNetwork == "" {
+		return nil, fmt.Errorf("default network not found")
+	}
+
+	// Get AppConfig for the selected network
+	initConfig, ok := networkManager.knownNetworks[defaultNetwork]
+	if !ok {
+		return nil, fmt.Errorf("selected network '%s' not found", defaultNetwork)
+	}
 
-func GetNetwork(network string) string {
-	//nolint:goconst
-	if network == "TESTNET" || network == "LABNET" || network == "BUILDNET" {
-		return network
+	appConfig := AppConfig{
+		NodeURL:    initConfig.URLs[0],
+		DNSAddress: initConfig.DNS,
+		Network:    defaultNetwork,
 	}
 
-	return "UNKNOWN"
+	networkManager.SetNetwork(appConfig)
+
+	return networkManager, nil
+}
+
+func (n *NetworkManager) SetNetworks(networks map[string]NetworkConfig) {
+	n.knownNetworks = networks
+}
+
+func (n *NetworkManager) Networks() *[]string {
```

Comment:
maybe this function should be name KnownNetworksName?

Reply:
i think its better to have short names and the comment is very explicit here : 	// Networks retrieves a pointer to a slice of known networks from the NetworkManager.
	// It returns a pointer to a slice containing the names of the known networks.
	// The slice will be updated if the known networks are modified.
	Networks() *[]string